File: Day_3/sol.js
class Solution {
    rearrange(arr) {
      let pos = [];
      let neg = [];
      for (let i = 0; i < arr.length; i++) {
        if (arr[i] < 0) {
          neg.push(arr[i]);
        } else {
          pos.push(arr[i]);
        }
      }
      let k = 0;
      let l = 0;
      for (let i = 0; i < arr.length; i++) {
        if (i % 2 === 0 && k < pos.length) {
          arr[i] = pos[k++];
        } else if (i % 2 !== 0 && l < neg.length) {
          arr[i] = neg[l++];
        } else if (k === pos.length && l < neg.length) {
          arr[i] = neg[l++];
        } else {
          arr[i] = pos[k++];
        }
      }
    }
  }

File: Day_17/sol.java
class Solution {
    public int minimizeCost(int k, int arr[]) {
        // Get the length of the array
        int n = arr.length;
        
        // Initialize a dynamic programming array with Integer.MAX_VALUE values
        int[] dp = new int[n];
        for (int i = 0; i < n; i++) {
            dp[i] = Integer.MAX_VALUE;
        }
        
        // The cost to reach the first stone is 0
        dp[0] = 0;

        // Iterate over each stone from the second stone to the last stone
        for (int i = 1; i < n; i++) {
            // Iterate over all possible previous stones that are within k steps
            for (int j = Math.max(0, i - k); j < i; j++) {
                // Calculate the minimum cost to reach the current stone
                dp[i] = Math.min(dp[i], dp[j] + Math.abs(arr[i] - arr[j]));
            }
        }

        // Return the minimum cost to reach the last stone
        return dp[n - 1];
    }
}